src/components/directory/viewer/dataModel/DataModel.vue

```diff
@@ -133,140 +100,108 @@ function onNodeCollapse(node: TreeNode) {
+    } as TreeNode;
+    if (property.property){
+      for (const [index,groupedProperty] of property.property.entries()){
+        createPropertyNode(groupedProperty,index,groupNode.children as TreeNode[],groupNode.key);
       }
     }
-    const typeEntities = await EntityService.getPartialEntities(iris, [RDF.TYPE]);
-    for (const typeEntity of typeEntities) {
-      node.children.forEach((child: TreeNode) => {
-        child.children!.forEach((child2: TreeNode) => {
-          if (child2.data.iri === typeEntity["@id"]) {
-            child2.data.typeIcon = getFAIconFromType(typeEntity[RDF.TYPE]);
-            child2.data.color = getColourFromType(typeEntity[RDF.TYPE]);
-          }
-        });
-      });
-    }
+
+    propertyList.push(groupNode);
   }
 }
 
-async function createOrNode(ttproperty: any, cardinality: string, index: any, propertyList: TreeNode[], parentKey: string) {
-  const property = {
-    key: parentKey + "-" + index.toString(),
-    label: "or",
-    children: [] as TreeNode[],
-    selectable: true,
-    loading: false,
-    data: {
-      cardinality: cardinality,
-      isOr: true,
-      type: [] as TreeNode[],
-      iri: ""
-    },
-    type: "or"
-  } as TreeNode;
-  for (const [index2, orProperty] of ttproperty[SHACL.OR].entries()) {
-    if (property.children && property.children.length < ttproperty[SHACL.OR].length) {
-      const type = orProperty[SHACL.CLASS] || orProperty[SHACL.NODE] || orProperty[SHACL.DATATYPE] || [];
-      const types = isArrayHasLength(type) ? type[0] : {};
-      const name = `${orProperty[SHACL.PATH]?.[0].name}  ${
-        isArrayHasLength(type) ? "(" + (type[0].name ? type[0].name : type[0]["@id"].slice(type[0]["@id"].indexOf("#") + 1)) + ")" : ""
-      }`;
-      property.data.typeIcon = getFAIconFromType(types);
-      property.data.color = getColourFromType(types);
-      const newChild = {
-        key: property.key + "-" + index2.toString(),
-        label: name,
-        children: [] as TreeNode[],
-        selectable: true,
-        type: "property",
-        loading: false,
-        data: {
-          iri: orProperty[SHACL.PATH][0]["@id"],
-          type: type
-        }
-      } as TreeNode;
-      property.children!.push(newChild);
-    }
-  }
-  for (const child of property.children!) {
-    child.loading = true;
-    for (const [index, type] of child.data.type.entries()) {
-      const newChild = {
-        key: child.key + "-" + index.toString(),
-        label: type.name,
-        children: [] as TreeNode[],
-        selectable: true,
-        type: "type",
-        loading: false,
-        data: {
-          iri: type["@id"]
-        }
-      } as TreeNode;
-      if (child.children && !child.children.includes(newChild)) child.children!.push(newChild);
+
+function createRangeNode(property: PropertyShape,propertyNode: TreeNode){
+  const ranges = property.clazz ? [property.clazz] : property.node ? property.node : property.datatype ? [property.datatype] : null;
```

Comment:
simplify for readability

Reply:
Don't understand the simplification request. which line is this?

Reply:
or did you mean 
  const ranges = property.clazz
      ? [property.clazz]
      : property.node || (property.datatype ? [property.datatype] : null);

Reply:
yes, I meant the nested ternary. better with ifs if there are multiple nested ternary operators

Reply:
actually it would be even better if you used '??'. if you are using the ternary only for null-checking, in js you can use the '??' shortcut. so the line above would be transformed to: 

`property.clazz ?? property.node ?? property.datatype ?? null`

Reply:
The issue is that property node is an array and the others are not so hence the conversion. Will replace with an if. This was from before so would have used an if myself from scratch because the idea of using ors as proxy for ifs doesnt fit well with me.